src/docs/development/platform-integration/c-interop.md

```diff
@@ -0,0 +1,202 @@
+---
+title: "Binding to native code using dart:ffi"
+description: "To use C code in your Flutter program, use the dart:ffi library (currently in preview)."
+---
+
+Flutter mobile can use the [dart:ffi][] library
+to call native C APIs. The _ffi_ acronym stands for
+[_foreign function interface._][FFI]
+Other terms for similar functionality include
+_native interface_ and _language bindings._
+
+{{ site.alert.note }}
+  The dart:ffi library is [in active development][ffi issue]
+  and isn't complete yet. Note that the API is likely to have
+  breaking changes between now and its completion.
+
+  Using the feature requires a Flutter 1.10.x
+  dev channel build. To switch to the dev channel and
+  upload the latest dev version, do the following:
+
+  ```terminal
+  $ flutter channel dev
+  $ flutter upgrade
+  ```
+  For more information on Flutter's channels,
+  see [Upgrading Flutter][].
+{{ site.alert.end }}
+
+API documentation is available from the Dart dev channel:
+[Dart API reference documentation][].
```

Comment:
I think we need a brief introduction to linking at be start:

A native library can be linked into an app either dynamically or statically. A statically linked library is embedded into the app's executable image, and will be loaded when the app starts.

Symbols from a statically linked library can be loaded via `DynamicLibrary.executable` or `DynamicLibrary.process`.

A dynamically linked library, by contrast, is distributed in a separate file or folder within the app, and loaded on-demand. On Android, a dynamically linked library is distributed as a set of `.so` (ELF) files, one for each architecture. On iOS, it's distributed as a `.framework` folder.

A dynamically linked library can be loaded into Dart via `DynamicLibrary.open`.